File: CS/Form1.cs
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraCharts;
using System.Collections;

namespace LegendMarkers {
    public partial class frmMain : Form {
        public frmMain() {
            InitializeComponent();
        }
        Hashtable legendMarkersTable = new Hashtable();
        void frmMain_Load(object sender, System.EventArgs e) {
                using (ChartControl fakeChart = (ChartControl)chart.Clone()) {
                    fakeChart.Legend.Visible = true;
                    fakeChart.BorderOptions.Visible = false;
                    fakeChart.Padding.All = 0;
                    fakeChart.BackColor = Color.Transparent;
                    fakeChart.Legend.Border.Visible = false;
                    fakeChart.Legend.Margins.All = 0;
                    fakeChart.Legend.Padding.All = 0;
                    fakeChart.Legend.TextVisible = false;
                    fakeChart.Legend.BackColor = Color.Transparent;
                    fakeChart.Legend.VerticalIndent = 0;
                    foreach (Series series in chart.Series)
                    {
                        Bitmap bitmap = GetMarkerImage(fakeChart, series);
                        legendMarkersTable.Add(series, bitmap);
                    }
                }
            gridControl1.DataSource = chart.Series;
        }
        Bitmap GetMarkerImage(ChartControl chart, Series series) {
            Bitmap bitmap = null;
            Series currentSeries = null;
            foreach (Series fakeSeries in chart.Series) {
                if (fakeSeries.Name == series.Name) {
                    currentSeries = fakeSeries;
                    fakeSeries.ShowInLegend = true;
                }
                else
                    fakeSeries.ShowInLegend = false;
            }
            if (currentSeries != null) {
                XYDiagram2D diagram = chart.Diagram as XYDiagram2D;
                if (diagram != null)
                    diagram.DefaultPane.Visible = false;
                SeriesViewColorEachSupportBase colorEachView = currentSeries.View as SeriesViewColorEachSupportBase;
                if (colorEachView != null && colorEachView.ColorEach)
                    chart.Size = new Size(chart.Legend.MarkerSize.Width, chart.Legend.MarkerSize.Height * currentSeries.Points.Count);
                else
                    chart.Size = chart.Legend.MarkerSize;
                bitmap = new Bitmap(chart.Size.Width, chart.Size.Height);
                chart.DrawToBitmap(bitmap, new Rectangle(Point.Empty, bitmap.Size));
            }
            return bitmap;
        }
      

        private void chart_ObjectSelected(object sender, HotTrackEventArgs e)
        {
            if (e.HitInfo.InSeries)
            {
                gridView1.FocusedRowHandle = gridView1.GetRowHandle(chart.Series.IndexOf(((Series)e.Object)));
                
            }
        }

        private void gridView1_CustomUnboundColumnData(object sender, DevExpress.XtraGrid.Views.Base.CustomColumnDataEventArgs e)
        {
            if (e.IsGetData && e.Column == gridColumn2)
            {

                e.Value = legendMarkersTable[((Series)e.Row)];
            }
        }

        private void gridView1_FocusedRowChanged(object sender, DevExpress.XtraGrid.Views.Base.FocusedRowChangedEventArgs e)
        {
            chart.SetObjectSelection(gridView1.GetRow(e.FocusedRowHandle));
        }
    }
}

File: VB/Form1.vb
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraCharts
Imports System.Collections

Namespace LegendMarkers

    Public Partial Class frmMain
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private legendMarkersTable As Hashtable = New Hashtable()

        Private Sub frmMain_Load(ByVal sender As Object, ByVal e As System.EventArgs)
            CreateLegendMarkers(chart)
            gridControl1.DataSource = chart.Series
        End Sub

        Private Sub CreateLegendMarkers(ByVal sourceChart As ChartControl)
            Using fakeChart As ChartControl = CType(sourceChart.Clone(), ChartControl)
                fakeChart.Legend.Visibility = DevExpress.Utils.DefaultBoolean.True
                fakeChart.BorderOptions.Visibility = DevExpress.Utils.DefaultBoolean.False
                fakeChart.Padding.All = 0
                fakeChart.BackColor = Color.Transparent
                fakeChart.Legend.Border.Visibility = DevExpress.Utils.DefaultBoolean.False
                fakeChart.Legend.Margins.All = 0
                fakeChart.Legend.Padding.All = 0
                fakeChart.Legend.TextVisible = False
                fakeChart.Legend.BackColor = Color.Transparent
                fakeChart.Legend.VerticalIndent = 0
                For Each series As Series In chart.Series
                    Dim bitmap As Bitmap = GetMarkerImage(fakeChart, series)
                    legendMarkersTable.Add(series, bitmap)
                Next
            End Using
        End Sub

        Private Function GetMarkerImage(ByVal chart As ChartControl, ByVal series As Series) As Bitmap
            Dim bitmap As Bitmap = Nothing
            Dim currentSeries As Series = Nothing
            For Each fakeSeries As Series In chart.Series
                If Equals(fakeSeries.Name, series.Name) Then
                    currentSeries = fakeSeries
                    fakeSeries.ShowInLegend = True
                Else
                    fakeSeries.ShowInLegend = False
                End If
            Next

            If currentSeries IsNot Nothing Then
                Dim diagram As XYDiagram2D = TryCast(chart.Diagram, XYDiagram2D)
                If diagram IsNot Nothing Then diagram.DefaultPane.Visibility = ChartElementVisibility.Hidden
                Dim colorEachView As SeriesViewColorEachSupportBase = TryCast(currentSeries.View, SeriesViewColorEachSupportBase)
                If colorEachView IsNot Nothing AndAlso colorEachView.ColorEach Then
                    chart.Size = New Size(chart.Legend.MarkerSize.Width, chart.Legend.MarkerSize.Height * currentSeries.Points.Count)
                Else
                    chart.Size = chart.Legend.MarkerSize
                End If

                bitmap = New Bitmap(chart.Size.Width, chart.Size.Height)
                chart.DrawToBitmap(bitmap, New Rectangle(Point.Empty, bitmap.Size))
            End If

            Return bitmap
        End Function

        Private Sub chart_ObjectSelected(ByVal sender As Object, ByVal e As HotTrackEventArgs)
            If e.HitInfo.InSeries Then
                gridView1.FocusedRowHandle = gridView1.GetRowHandle(chart.Series.IndexOf(CType(e.Object, Series)))
            End If
        End Sub

        Private Sub gridView1_CustomUnboundColumnData(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.CustomColumnDataEventArgs)
            If e.IsGetData AndAlso e.Column Is colImage Then
                e.Value = legendMarkersTable(CType(e.Row, Series))
            End If
        End Sub

        Private Sub gridView1_FocusedRowChanged(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.FocusedRowChangedEventArgs)
            chart.SetObjectSelection(gridView1.GetRow(e.FocusedRowHandle))
        End Sub
    End Class
End Namespace
